Windows network drives not always matched by `net use` regex

### Describe the bug

The Windows network drive check implemented in [this PR](https://github.com/vitejs/vite/pull/12580/files) uses the following regular expression:

https://github.com/vitejs/vite/blob/63217529154dc6dc53d5861a068e80a1aec0fdb3/packages/vite/src/node/utils.ts#L617

This assumes a non-empty status from the `net use` command, which is not always the case. I don't know what causes certain valid mapped drives to not show up with a status, but you can see an example in [this article](https://lazyadmin.nl/it/net-use-command/) although it's not called out.

Because of this, Vite cannot always correctly resolve paths for valid mapped network drives. Making the status check in the regex optional fixes the issue:

`const parseNetUseRE = /^(\w+)? +(\w:) +([^ ]+)\s/`

### Reproduction

not possible with repo alone

### Steps to reproduce

Create a mapped network drive that gives no status when checked with `net use`. (No idea why this happens or how to do it on purpose.) Attempt to run Vite from that mapped drive.

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (6) x64 Intel(R) Xeon(R) CPU E7-4870 v2 @ 2.30GHz
    Memory: 5.69 GB / 12.00 GB
  Binaries:
    Node: 18.1.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.8.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (111.0.1661.41)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    vite: 4.3.0 => 4.3.0
```


### Used Package Manager

npm

### Logs

<details>
<summary>Click to expand!</summary>

```shell
failed to load config from <network drive>:<repo path>\vite.config.js
error when starting dev server:
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '<network drive>:<!!network drive path, repeated!!><repo path>\node_modules\svelte\store\index.mjs' imported from <network drive>:<repo path>\vite.config.js.timestamp-1682168805046-b61e4fd57df9b.mjs
    at new NodeError (node:internal/errors:377:5)
    at finalizeResolution (node:internal/modules/esm/resolve:405:11)      
    at moduleResolve (node:internal/modules/esm/resolve:966:10)
    at defaultResolve (node:internal/modules/esm/resolve:1174:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:605:30)        
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:318:18)   
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:80:40)    at link (node:internal/modules/esm/module_job:78:36)
```
</details>

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to [vuejs/core](https://github.com/vuejs/core) instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vite/src/node/utils.ts
+++ b/packages/vite/src/node/utils.ts
@@ -574,7 +574,7 @@ function windowsMappedRealpathSync(path: string) {
   }
   return realPath
 }
-const parseNetUseRE = /^(\w+) +(\w:) +([^ ]+)\s/
+const parseNetUseRE = /^(\w+)? +(\w:) +([^ ]+)\s/
 let firstSafeRealPathSyncRun = false
 
 function windowsSafeRealPathSync(path: string): string {

